src/Main.hs

```diff
@@ -38,9 +45,94 @@ printIssue = BSL.putStr . (<> "\0") . encode
 
 --------------------------------------------------------------------------------
 
-shFiles :: [FilePath] -> IO [FilePath]
-shFiles paths =
-  fmap concat $! sequence $! fmap (matched . globDir [compile "**/*.sh"]) paths
+shellScripts :: [FilePath] -> IO [FilePath]
+shellScripts paths = do
+  dotShFiles <- concat . fst <$> globDir patterns "."
+  allScripts <- filterM validateScript $ dotShFiles ++ otherFiles
+  return $ fmap clean allScripts
   where
-    matched :: Functor f => f ([[a]], [b]) -> f [a]
-    matched x = (concat . fst) <$> x
+    (dirs, otherFiles) = partition hasTrailingPathSeparator paths
+
+    clean :: String -> String
+    clean ('.' : '/' : x) = x
+    clean x               = x
+
+    patterns :: [Pattern]
+    patterns = fmap (compile . (++ "**/*.sh")) dirs
```

Comment:
@filib I've only seen regular expressions that work for this. Match for all files that do not contain a '.'